solver/src/main/java/org/chocosolver/solver/constraints/binary/PropSquare.java

```diff
@@ -176,24 +205,12 @@ protected boolean updateHolesinY() throws ContradictionException {
                     vrms.add(value);
                 }
             }
-            return vars[1].removeValues(vrms, this);
-        } else if (vars[1].hasEnumeratedDomain()) {
-            int lb = vars[1].getLB();
-            int ub = vars[1].getUB();
-            while (!vars[0].contains(sqr(lb))) {
-                lb = vars[1].nextValue(lb);
-                if (lb > ub) break;
-            }
-            boolean filter = vars[1].updateLowerBound(lb, this);
-
-            while (!vars[0].contains(sqr(ub))) {
-                ub = vars[1].nextValue(ub);
-                if (ub < lb) break;
-            }
-            return filter | vars[1].updateUpperBound(ub, this);
+            impact |= vars[1].removeValues(vrms, this);
         }
-        return false;
+        return impact;
     }
 
-
+    private boolean hasHoles(IntVar var) {
+        return (var.getUB() - var.getLB() + 1) > var.getDomainSize();
+    }
```

Comment:
@ArthurGodet you suggest to remove it because the following code is expected to create holes everywhere?
```
// check perfect squares once and for all
            int ub = vars[0].getUB();
            vrms.clear();
            vrms.setOffset(vars[0].getLB());
            for (int value = vars[0].getLB(); value <= ub; value = vars[0].nextValue(value)) {
                if (!MathUtils.isPerfectSquare(value)) {
                    vrms.add(value);
                }
            }
            impact = vars[0].removeValues(vrms, this);
```

Reply:
Yes, indeed. As long as vars[0] has an enumerated domain and its bounds are different, it will necessarily contains holes because of this code.